R/import_exports.R

```diff
@@ -0,0 +1,194 @@
+#' Total exports by sectors
+#'
+#' This function returns total exports by sectors in the  Dominican Republic
+#' based on the specified frequency.
+#'
+#' @param frecuencia A character string that specifies the frequency of the
+#' data to be downloaded. Valid options are "mensual",
+#' "trimestral",  or "anual".
+#'
+#' @return A data frame
+#' @export
+#'
+#' @examples
+#' get_exportaciones("mensual")
+#' get_exportaciones("trimestral")
+#' get_exportaciones("anual")
+
+get_exportaciones <- function(frecuencia = "mensual") {
+  checkmate::assert_choice(
+    frecuencia,
+    choices = c("mensual", "trimestral", "anual"))
+
+  periodo <- c(2010:lubridate::year(Sys.Date()))
+
+  url_descarga <- paste0(
+    "https://cdn.bancentral.gov.do/documents/estadisticas/",
+    "sector-externo/documents/Exportaciones_Mensuales_",
+    periodo, "_6.xls?v=1571253704905"
+  )
+
+  files_path <- purrr::map_chr(
+    periodo,
+    ~tempfile(pattern = "", fileext = ".xls")
+  )
+
+  dw_check <- purrr::possibly(utils::download.file, otherwise = -1) # nolint
```

Comment:
Excelente uso de possibly para evitar que el error al intentar descargar un archivo que no existeEntiendo detenga la ejecusión de la función. 

También comprendo la lógica que quiere implementar usando `otherwhise = -1`, ahora bien, creo que hay mejores alternativas, más claras para mantener la función en el futuro. Paso a expandir mi opinión:

1. usar `otherwise = NA`
2. descargar los archivos usando `purrr::walk2`
3. usar `file.exists` para comprobar si efectivamente los archivos se descargaron y filtrar con eso

Te quedaría algo como sigue:

```r
  safe_download <- purrr::possibly(utils::download.file, otherwise = NA)

  purrr::walk2(
    url_descarga,
    files_path,
    \(url, file) safe_download(url, file, mode = "wb", quiet = TRUE)
  ) |>
    suppressWarnings()

  files_path <- files_path[file.exists(files_path)]
```